subscriptions-api-openapi.yaml

```diff
@@ -125,126 +132,349 @@ paths:
+                    - $ref: "#/components/schemas/DialectFilter"
+          additionalProperties: false
+    ExactFilter:
+      allOf:
+        - $ref: "#/components/schemas/Filter"
+        - type: object
+          description: "all filter" 
+          properties:
+            exact:
+              title: "exact filter"
+              description: "This filter evaluates to 'true' if the 'value' exactly matches the value of the indicated CloudEvents context attribute"
+              required:
+                - attribute
+                - value
+              properties:
+                attribute:
+                  type: string
+                  description: attribute name
+                value:
+                  type: string
+                  description: value
+          additionalProperties: false
+    PrefixFilter:
+      allOf:
+        - $ref: "#/components/schemas/Filter"
+        - type: object
+          description: "all filter" 
+          properties:
+            prefix:
+              title: "prefix filter"
+              description: "This filter evaluates to 'true' if the 'value' is a prefix of the value of the indicated CloudEvents context attribute"
+              required:
+                - attribute
+                - value
+              properties:
+                attribute:
+                  type: string
+                  description: attribute name
+                value:
+                  type: string
+                  description: value
+              additionalProperties: false
+    SuffixFilter:
+      allOf:
+        - $ref: "#/components/schemas/Filter"
+        - type: object
+          description: "all filter" 
+          properties:
+            suffix:
+              title: "suffix filter"
+              description: "This filter evaluates to 'true' if the 'value' is a suffix of the value of the indicated CloudEvents context attribute"
+              required:
+                - attribute
+                - value
+              properties:
+                attribute:
+                  type: string
+                  description: attribute name
+                value:
+                  type: string
+                  description: value
+              additionalProperties: false
+    DialectFilter:
+      allOf:
+        - $ref: "#/components/schemas/Filter"
+        - type: object
+          minProperties: 1
+          maxProperties: 1
+          additionalProperties:
+            title: "dialect filter"
+            description: "This filter evaluates to 'true' based on the rules of specific dialect indicated by the key of the filter"
+            required:
+              - expression
+            properties:
+              attribute:
+                type: string
+                description: attribute name
+              expression:
+                type: string
+                description: value
+    ProtocolSettings:
+      type: object
+      additionalProperties: false
```

Comment:
This doesn't apply to the "inheritors", as explained [here](https://json-schema.org/draft/2019-09/json-schema-core.html#additionalProperties):

> The behavior of this keyword depends on the presence and annotation results of "properties" and "patternProperties" within the same schema object.

The _schema object_ is literally the schema where the `additionalProperties` keyword lives (`allOf` doesn't mean **merge all the schemes** for the spec)

You need to have `additionalProperties: false` in HTTPSettings, MQTTSettings, ... otherwise a validator reject all type of settings.